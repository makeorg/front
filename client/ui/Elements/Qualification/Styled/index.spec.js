import React from 'react';
import renderer from 'react-test-renderer';
import { QualifyButtonStyle } from './index';

describe('QualifyButtonStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<QualifyButtonStyle />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must set color from props', () => {
    const component = renderer
      .create(<QualifyButtonStyle color="red" />)
      .toJSON();
    expect(component).toHaveStyleRule('color', 'red');
    expect(component).not.toHaveStyleRule('background-color', 'red', {
      modifier: ':focus',
    });
    expect(component).not.toHaveStyleRule('background-color', 'red', {
      modifier: ':active',
    });
  });
});
