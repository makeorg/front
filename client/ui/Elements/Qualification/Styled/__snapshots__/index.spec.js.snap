// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyle snapshot by default 1`] = `
.c0 {
  font-family: 'Roboto Bold',Arial,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  border-width: 2px;
  font-size: 12px;
  line-height: 26px;
  border-style: solid;
  padding: 0 10px;
  border-radius: 36px;
}

@media (min-width:767px) {
  .c0 {
    font-size: 14px;
    line-height: 28px;
  }
}

<button
  className="c0"
/>
`;

exports[`QualifyButtonStyle snapshot by default 1`] = `
.c0 {
  font-family: 'Roboto Bold',Arial,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  border-width: 2px;
  font-size: 12px;
  line-height: 26px;
  border-style: solid;
  padding: 0 10px;
  border-radius: 36px;
  background-color: rgb(255,255,255);
}

@media (min-width:767px) {
  .c0 {
    font-size: 14px;
    line-height: 28px;
  }
}

<button
  className="c0"
/>
`;
